### ISSUE TITLE
wasm32 assertion failure in writePatchableLEB()

### ISSUE THREAD

**Issue by nikic** [llvm/llvm-project#54012]
```
target triple = "wasm32-unknown-unknown"

@c = private constant i8* null

define i8* @test() {
  %v = load i8*, i8** getelementptr (i8*, i8** @c, i32 -1), align 4
  ret i8* %v
}
```

With `llc -filetype=obj`, this results in an assertion failure:

```
llc: /home/npopov/repos/llvm-project/llvm/lib/MC/WasmObjectWriter.cpp:146: void {anonymous}::writePatchableLEB(llvm::raw_pwrite_stream&, uint64_t, uint64_t) [with int W = 5; uint64_t = long unsigned int]: Assertion `SizeLen == W' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: build/bin/llc -filetype=obj
[...]
 #8 0x00000000028392cb (anonymous namespace)::WasmObjectWriter::applyRelocations(llvm::ArrayRef<(anonymous namespace)::WasmRelocationEntry>, unsigned long, llvm::MCAsmLayout const&) WasmObjectWriter.cpp:0:0
 #9 0x000000000283e87e (anonymous namespace)::WasmObjectWriter::writeOneObject(llvm::MCAssembler&, llvm::MCAsmLayout const&, (anonymous namespace)::WasmObjectWriter::DwoMode) WasmObjectWriter.cpp:0:0
#10 0x000000000283f1d7 (anonymous namespace)::WasmObjectWriter::writeObject(llvm::MCAssembler&, llvm::MCAsmLayout const&) WasmObjectWriter.cpp:0:0
#11 0x00000000027ba976 llvm::MCAssembler::Finish() (build/bin/llc+0x27ba976)
#12 0x0000000001fb88fb llvm::AsmPrinter::doFinalization(llvm::Module&) (build/bin/llc+0x1fb88fb)
#13 0x00000000026f4ddd llvm::FPPassManager::doFinalization(llvm::Module&) (build/bin/llc+0x26f4ddd)
#14 0x000000000270099f llvm::legacy::PassManagerImpl::run(llvm::Module&) (build/bin/llc+0x270099f)
#15 0x0000000000674f88 main (build/bin/llc+0x674f88)
```

Of course, the input IR is UB, but it probably shouldn't crash?

**Comment by sbc100** [comment]
Proposed fix is here: https://reviews.llvm.org/D120522

**Comment by sbc100** [comment]
@nikic are you able to take care of that cherry pick?

**Comment by nikic** [comment]
I think this is just missing a dependent revision that modified the test...

/cherry-pick db7b1af8ef8f59600cbb6e529164acdbe4c8f0a2 4c75521ce0b1ac55fba1a91ef5156fc811f5dfcb

**Comment by tstellar** [comment]
@sunfishcode What do you think about backporting these? https://github.com/llvm/llvm-project/commit/db7b1af8ef8f59600cbb6e529164acdbe4c8f0a2 https://github.com/llvm/llvm-project/commit/4c75521ce0b1ac55fba1a91ef5156fc811f5dfcb

**Comment by nikic** [comment]
I don't think that @sunfishcode is still active in LLVM, it would probably be good if one of the current WebAssembly maintainers (@aheejin @tlively @pmatos @sbc100 maybe?) could pick a new code owner and add them in https://github.com/llvm/llvm-project/blob/1bd33691cb1d10fb118ddf7f9fb305b112f67cbe/llvm/CODE_OWNERS.TXT#L90-L92.

**Comment by tstellar** [comment]
Merged: f6a4df3a0a3d e6d2aa9b0f13
